app/src/main/java/com/kdani/dagger2_11/di/component/AppComponent.java

```diff
@@ -13,7 +13,7 @@
 
 @Singleton
 @Component(modules = { AppModule.class,
-        ActivityModule.class,
+        MainActivityModule.class,
```

Comment:
Also, here i wanted to have 1 single module for all of my activities. I'm not sure if if we'll be able to with this approach.

Reply:
I don't think you will be able to do that. The best example for you to follow is this one: https://github.com/googlesamples/android-architecture/tree/todo-mvp-dagger/

Reply:
i was following this that's why i created activityBindingModule from [here](https://github.com/googlesamples/android-architecture/blob/todo-mvp-dagger/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/ActivityBindingModule.java)

Reply:
Actually - looking at that one seems like I made a mistake. I'll have another go and if I can get it working it might be closer to what you want